framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -2230,24 +2241,21 @@ AutomaticMortarGeneration::secondariesToMortarSegments(const Node & node) const
     mooseError("end of secondary map in associatedMortarSegments");
 
   const auto & secondary_elems = secondary_it->second;
-  std::vector<MortarFilterIter> ret;
-  ret.reserve(secondary_elems.size());
+  std::vector<MortarFilterIter> ret(secondary_elems.size());
 
   for (const auto i : index_range(secondary_elems))
   {
     auto * const secondary_elem = secondary_elems[i];
     auto msm_it = _secondary_elems_to_mortar_segments.find(secondary_elem);
     if (msm_it == _secondary_elems_to_mortar_segments.end())
-      // We may have removed this element key from this map
-      continue;
-
+      mooseError("end of mortar segment map in associatedMortarSegments");
```

Comment:
Each secondary element in the maps must now always have a corresponding mortar segment by mesh construction (my understanding).

Reply:
We are removing the secondaries from the nodal map if the msm is empty: Is that general enough to enforce this error?

Reply:
Yes it should be. We should not have any secondaries without msms either in the node-to-secondary map or the secondary-to-msm map.

Reply:
We may be missing something. I created a typical bouncing block example with a secondary sideset that wraps around its surface and came across that error.

Reply:
Haha look at you coming up with more tricky tests! This is good. If you can, please push up that test, and I will take a look

Reply:
Just did!